src/main/java/io/cryostat/platform/PlatformClient.java

```diff
@@ -55,6 +59,19 @@ default void load(Promise<EnvironmentNode> promise) {
 
     List<ServiceRef> listDiscoverableServices();
 
+    default List<ServiceRef> listUniqueReachableServices() {
+        return listDiscoverableServices().stream()
+                .filter((ref) -> ref.getJvmId() != null)
+                .collect(
```

Comment:
I think there's a form like this that might be a bit easier to read and perhaps do fewer allocations:

```java
return listDiscoverableServices().stream()
  .filter((ref) -> ref.getJvmId() != null)
  .sorted((ref1, ref2) -> ref1.getJvmId().compareTo(ref2.getJvmId())
  .toList()
  .sorted
```